File: AEDS/Java/Trabalho Agenda/src/Agenda/PessoaFisica.java
package Agenda;

public class PessoaFisica extends Pessoa {

    private static final String validacao = "^[0-9]{3}\\.?[0-9]{3}\\.?[0-9]{3}\\-?[0-9]{2}$";

    public PessoaFisica(String numero) throws IllegalArgumentException{

        if (validaCPF(numero)) this.CadastroPessoa = numero;

        else throw new IllegalArgumentException("Invalid CPF format");
    }

    @Override
    public void setNumero(String CadastroPessoa) { this.CadastroPessoa = CadastroPessoa; }

    private boolean validaCPF(String numero){

        if (!numero.matches(validacao)){

            return false;
        }

        return true;

    }
}

File: AEDS/C#/Agenda em CSharp/PessoaFisica.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.RegularExpressions;

namespace Agenda_em_CSharp
{
    class PessoaFisica : Pessoa{

        private static readonly string validador = "^[0-9]{3}\\.?[0-9]{3}\\.?[0-9]{3}\\-?[0-9]{2}$";

        public PessoaFisica(string numero)
        {
            if (ValidaCPF(numero)) this.numero = numero;

            else throw new ArgumentException("Invalid format");
        }

        private bool ValidaCPF(string numero)
        {

            if (Regex.IsMatch(numero, validador)) return true;

            else return false;

        }
        public override void SetNumero(string numero) {

            if (ValidaCPF(numero)) this.numero = numero;

            else throw new ArgumentException("Invalid CPF");
        }
    }
}
